### PR TITLE
update install scripts

### PR DESCRIPTION
**Pull request by prasadtalasila**
Updates the install scripts with the following improvements.

* Tomcat 8.5 installation with the URL coming from Apache EU mirror
* Updates node.js to 8.9x
* Removes errors in the tomcat.service file
* Improves the code in install.sh shell script

@raghavagrawal please review the changes proposed here.

### CODE REVIEW

File: scripts/install.sh [hunk 1]
```diff
@@ -3,73 +3,80 @@
...
-sudo apt-get -y install -y nodejs
-sudo apt-get -y install build-essential
+sudo apt-get update && sudo apt-get install --force-yes -y nodejs build-essential
 rm nodesource_setup.sh
 
+#install npm modules for browser client dependencies
 mkdir -p src/main/webapp/WEB-INF/node_modules
-#get the npm modules for js files of webpages
-yes | npm install --no-bin-links src/main/webapp/WEB-INF
+npm install --prefix src/main/webapp/WEB-INF
```
Comment (raghavagrawal) [inline scripts/install.sh]: @prasadtalasila I was facing problems in installation when using --prefix. I request you see this issue.
[https://github.com/npm/npm/issues/9901](url)

File: scripts/install.sh [hunk 2]
```diff
@@ -3,73 +3,80 @@
...
-mkdir /opt/tomcat
-tar xzvf apache-tomcat-8*tar.gz -C /opt/tomcat --strip-components=1
-#Give the tomcat group ownership over the entire installation directory:
-sudo chgrp -R tomcat /opt/tomcat
+sudo groupadd tomcat
+sudo useradd -s /bin/false -g tomcat -d /opt/tomcat tomcat
+
+#download and install tomcat
+curl -O http://www-eu.apache.org/dist/tomcat/tomcat-8/v8.5.27/bin/apache-tomcat-8.5.27.tar.gz
+sudo tar xzf apache-tomcat-8*tar.gz -C /opt/tomcat --strip-components=1
```
Comment (raghavagrawal) [inline scripts/install.sh]: @prasadtalasila We will have to make /opt/tomcat directory.

File: scripts/install.sh [hunk 3]
```diff
@@ -3,73 +3,80 @@
...
-chmod 777 /opt/darshini-es/logs
-mkdir -p /opt/darshini-logs
-chmod 777 /opt/darshini-logs
-touch /opt/darshini-logs/darshini
-chown tomcat:tomcat /opt/darshini-logs/darshini
-chmod 777 /opt/darshini-logs/darshini
-
-#install nodejs
-curl -sL https://deb.nodesource.com/setup_6.x -o nodesource_setup.sh
+sudo mkdir -p /opt/darshini-es/data
```
Comment (raghavagrawal) [inline scripts/install.sh:68]: @prasadtalasila Will these 'sudo' be required as we will be running the scripts as superuser.

**Comment by prasadtalasila** [comment]
@adityaketkar can you see any reason for the reduction in code coverage in this PR?

**Comment by adityaketkar** [comment]
@prasadtalasila  I think there is something wrong with codecov report. 
Please see the screenshot below : 
image


After I going through the commit diff's, I am pretty sure that the src/main/* files are untouched by those commits. But still, the report suggests that lines were added to the branch.
Is it possible to rerun the analysis?

**Comment by prasadtalasila** [comment]
@adityaketkar I rebased the tomcat8 branch with the dev branch. Now the codecov is showing a correct report. Please verify the same.

**Review by raghavagrawal — CHANGES_REQUESTED** [review]
@prasadtalasila I have added a few comments for your approval

File: scripts/install.sh (hunk 3 above)
Comment (prasadtalasila) [reply scripts/install.sh:68]: We could do it either way. I thought that running the script as regular user, but giving sudo to the required commands would be better from security point of view.

Comment (prasadtalasila) [reply scripts/install.sh:68]: Also the [install](https://github.com/prasadtalasila/BITS-Darshini/wiki/Install) document has been updated accordingly.

**Comment by prasadtalasila** [comment]
@raghavagrawal one unaddressed issue is the use of `--prefix`. You said that you were facing problems. Can you please elaborate on the problems and give a log?

**Pull request merged**